### PR TITLE
AEA-3708 Adds tests for UserRoleServiceV2

### PR DESCRIPTION
**Pull request by JackSpagnoliNHS**
## Summary
* Routine Change
* :sparkles: New Feature

Adds tests for the UserRoleServiceV2 sharedflow.

Tests are in 2 parts, one against a deployment of the flow with a default header name, and one using a custom header name. Currently these point against the TestUserRoleService sharedflow, which will need changing once UserRoleServiceV2 is deployed.

A change to AssignMessage.Errors.CatchAllMessage policy and the test_user_roles tests has been made to remove the fhir formatting of responses as UserRoleServiceV2 does this internally and the previous implementation failed to json deserialise.

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: proxies/live/apiproxy/policies/FlowCallout.UserRoleServiceV2.CustomHeader.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<FlowCallout async="false" continueOnError="false" enabled="true" name="FlowCallout.UserRoleServiceV2.CustomHeader">
+    <DisplayName>FlowCallout.UserRoleServiceV2.CustomHeader</DisplayName>
+    <FaultRules/>
+    <Properties/>
+    <Parameters>
+        <Parameter name="user_role_header">NHSD-URID</Parameter>
+    </Parameters>
+    <!-- <SharedFlowBundle>UserRoleServiceV2</SharedFlowBundle> -->
+    <SharedFlowBundle>TestUserRoleService</SharedFlowBundle>
```
Comment (acarriedev) [inline proxies/live/apiproxy/policies/FlowCallout.UserRoleServiceV2.CustomHeader.xml:10]: Need change when deployed the official one

File: proxies/live/apiproxy/policies/FlowCallout.UserRoleServiceV2.DefaultHeader.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<FlowCallout async="false" continueOnError="false" enabled="true" name="FlowCallout.UserRoleServiceV2.DefaultHeader">
+    <DisplayName>FlowCallout.UserRoleServiceV2.DefaultHeader</DisplayName>
+    <FaultRules/>
+    <Properties/>
+    <!-- <SharedFlowBundle>UserRoleServiceV2</SharedFlowBundle> -->
+    <SharedFlowBundle>TestUserRoleService</SharedFlowBundle>
```
Comment (acarriedev) [inline proxies/live/apiproxy/policies/FlowCallout.UserRoleServiceV2.DefaultHeader.xml:7]: Change when official one deployed (remove this one manually too)

File: tests/test_user_role_service_v2.py [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+
+def set_custom_header_name(headers):
+    default_header = headers.get("NHSD-Session-URID")
+    return {"NHSD-URID": default_header}
+
+
+class TestUserRoles:
+    """A test suite for testing userrole in id tokens/headers/userinfo"""
+
+    happy_path_params = [
```
Comment (acarriedev) [inline tests/test_user_role_service_v2.py:13]: I think these would be better as constants out of the class coz of the way they have to be accessed in parametrize. Just a style/preference thing, functionality is good

**Review by acarriedev — CHANGES_REQUESTED** [review]
Looks good, with a few comments to address. Also there is a linting thing that the PR isn't liking about branch names. You'll need to add aea to the list here - https://github.com/NHSDigital/shared-flow-testing/blob/ecc39d65d0c7bc62a28d4bec37010afd9c64d6a8/.github/workflows/pr-lint.yaml#L8

File: tests/test_user_role_service_v2.py (hunk 3 above)
Comment (JackSpagnoliNHS) [reply tests/test_user_role_service_v2.py]: da70d9960e5b8497073bb89c750283be070ecf71

**Review by acarriedev — APPROVED** [review]

File: proxies/live/apiproxy/policies/FlowCallout.UserRoleServiceV2.DefaultHeader.xml (hunk 2 above)
Comment (JackSpagnoliNHS) [reply proxies/live/apiproxy/policies/FlowCallout.UserRoleServiceV2.DefaultHeader.xml]: 66cd8d02c1f8f688065db44c635e217948744eab

File: proxies/live/apiproxy/policies/FlowCallout.UserRoleServiceV2.CustomHeader.xml (hunk 1 above)
Comment (JackSpagnoliNHS) [reply proxies/live/apiproxy/policies/FlowCallout.UserRoleServiceV2.CustomHeader.xml]: 66cd8d02c1f8f688065db44c635e217948744eab

**Review by acarriedev — APPROVED** [review]

**Pull request merged**